### DISCUSSION TITLE
Polygon edges do not intersect error

### DISCUSSION THREAD

**Discussion by mireiattt** [cduck/hyperbolic#13] [Q&A] [answered]
When drawing individual tile edges, after a depth of 25 I get an error that the polygon edges do not intersect. Could this be a rounding error of the package? When I use infinite lines that have segment=False, I do not get this error anymore. 

The error occurs as the line function calls polygon formation:
for m in range(len(tile.vertices)):
             d = Line.fromPoints(*tile.vertices[m%Integer(3)], *tile.vertices[(m+Integer(1))%Integer(3)], segment=True).toDrawables(elements, hwidth= RealNumber('0.02'), fill=color, **kwargs)
              ds.extend(d)

Then in Hypercycle, toDrawables 
poly = poincare.Polygon.fromEdges(edges, join=True)
 is called 

and then Polygon, fromEdges is called and at the following line:
return Polygon(edges=edges, join=join)

the error is raised: polygon edges do not intersect. 

But I am just trying to draw a line from 2 points, the edges do not need to intersect because there is only one edge, how do I fix this?

**Comment by cduck** [comment, marked as answer]
This sounds like a rounding error.  Are you using `Integer` and `RealNumber` to avoid rouding errors?  This library hasn't been designed with special number types in mind so that may or may not help.

The reason you get the polygon error is because the `hwidth` parameter converts your line into a hyperbolic "rectangle" before drawing it.  Rounding error must be causing the computed sides of this rectangle to not quite intersect.  One end of the line may be very close to the boundary of the Poincare disk.

Why are you drawing polygons up to depth 25?  Anything after 15 or so will be extremely tiny (smaller than a pixel) depending on the rendered resolution.

**Reply by mireiattt** [reply]
Integer and RealNumber are the way my Jupyter notebook is reproducing my inputs. Inkspace allows for a higher resolution and works nicely with higher depths. Is there a way to get around this rounding error?

**Reply by cduck** [reply]
It wouldn't be easy.  You would have to check every line of the package code to make sure it doesn't accidentally introduce `float` type numbers.
